core/src/test/java/org/mobilitydata/gtfsvalidator/validator/TableHeaderValidatorTest.java

```diff
@@ -71,13 +75,32 @@ public void missingRequiredColumnShouldGenerateNotice() {
             new CsvHeader(new String[] {"stop_name"}),
             ImmutableSet.of("stop_id", "stop_name"),
             ImmutableSet.of("stop_id"),
+            Set.of(),
             container);
 
     assertThat(container.getValidationNotices())
         .containsExactly(new MissingRequiredColumnNotice("stops.txt", "stop_id"));
     assertThat(container.hasValidationErrors()).isTrue();
   }
 
+  @Test
+  public void missingRecommendedColumnShouldGenerateNotice() {
+    NoticeContainer container = new NoticeContainer();
+    new DefaultTableHeaderValidator()
+        .validate(
+            "stops.txt",
+            new CsvHeader(new String[] {"stop_name"}),
+            Set.of("stop_id", "stop_name"),
+            Set.of(),
+            Set.of("stop_id"),
+            container);
+
+    assertThat(container.getValidationNotices())
+        .containsExactly(new MissingRecommendedColumnNotice("stops.txt", "stop_id"));
+    assertThat(container.hasValidationErrors()).isFalse();
+    assertThat(container.hasValidationWarnings()).isTrue();
```

Comment:
These conditions are already tested by line 99, as the container only holds the MissingRecommendedColumnNotice

Reply:
Right. I might as well remove something similar in missingRequiredColumnShouldGenerateNotice that I just copied and slightly modified. Line 83.